src/sota_tools/garage_push.cc

```diff
@@ -147,6 +151,27 @@ int main(int argc, char **argv) {
         return EXIT_FAILURE;
       }
     }
+
+    if (manifest_path != "") {
+      std::string manifest_json;
+      manifest_json = xml2json(Utils::readFile(manifest_path).c_str());
```

Comment:
I'd originally assumed we could just make a call to `repo manifest -r` from the shell, but did you have a particular motivation to read it from file? Presumably in meta-updater you'd just reuse the file created by the manifest class?

Reply:
> Presumably in meta-updater you'd just reuse the file created by the manifest class?

Yes, that's what I opted to do. In general it's more flexible to pass a complete path, instead of depending on being able to execute `repo` from where garage-push is launched.

Reply:
Makes sense the more I think about it. Thanks!